### DISCUSSION TITLE
How to target minimap parts via CSS?

### DISCUSSION THREAD

**Discussion by jagthedrummer** [katspaugh/wavesurfer.js#3330] [Q&A]
I'm trying to style some of the subparts of the minimap and I can't seem to target them at all via CSS. I can target the `minimap` part itself, but I can't figure out how to target the `wrapper` or `progress` parts of the minimap.

I'm trying to add a bit of padding inside the "you are here" bar so that it's not right up against the waveform.

In the main player I did this by adding some padding to the `wrapper` and `progress` parts.

Before:

![CleanShot 2023-11-09 at 10 04 27@2x](https://github.com/katspaugh/wavesurfer.js/assets/58702/42f61380-18c3-481e-9b7e-0b0eb04fcc87)

After adding this CSS:

```css
.wavesurfer ::part(wrapper), .wavesurfer ::part(progress){
  padding-top:1em;
}
```
![CleanShot 2023-11-09 at 10 06 19@2x](https://github.com/katspaugh/wavesurfer.js/assets/58702/2f40bef1-9e45-4c00-a70a-644c2f14f864)

I would have thought that the CSS above would have applied to the minimap as well since it uses parts that are named the same, but it looks like there are nested shadow DOMs involved and it seems like you just can't target two levels down into shadow DOMs.

I've been able to add a little padding to the top of the minimap itself by targeting it directly, but the cursor line only goes as high as the top of the waveform, so it would look nicer if the padding were on the `wrapper` instead of on the `minimap` part.

```CSS
.wavesurfer ::part(minimap) {
  margin-bottom:1em;
  padding-top:1em;
}
```

![CleanShot 2023-11-09 at 10 10 31@2x](https://github.com/katspaugh/wavesurfer.js/assets/58702/e8d526f5-3d5f-4af9-b067-2e1b4e686a98)

Am I missing something? Is there a way to target those parts?

Thanks for the awesome library!

**Comment by katspaugh** [comment]
You could use the `barHeight` option and set it to something lower than 1. E.g. `barHeight: 0.8` would squeeze the waveform vertically by 20%, giving you empty "padding" around it on the canvas. And then increase the `height` option accordingly to stretch it back to your desired size.

**Reply by jagthedrummer** [reply]
Thank you, that did the trick! My initial height for the minimap was `12` and I ended up setting `height: 30, barHeight: 0.4` to get this:

![CleanShot 2023-11-09 at 12 12 45](https://github.com/katspaugh/wavesurfer.js/assets/58702/68ed478b-6349-4745-a2a8-0d7884eec6dd)
